tests/integ/sagemaker/workflow/test_model_steps.py

```diff
@@ -0,0 +1,1138 @@
+            assert step["StepStatus"] == "Succeeded"
+            if _CREATE_MODEL_NAME_BASE in step["StepName"]:
+                assert step["Metadata"][_CREATE_MODEL_TYPE]
+        assert len(execution2_steps) == 3
+    finally:
+        try:
+            pipeline.delete()
+        except Exception:
+            pass
+
+
+def test_mxnet_model_registration(
+    pipeline_session,
+    role,
+    cpu_instance_type,
+    pipeline_name,
+):
+    base_dir = os.path.join(DATA_DIR, "mxnet_mnist")
+    source_dir = os.path.join(base_dir, "code")
+    entry_point = os.path.join(source_dir, "inference.py")
+    mx_mnist_model_data = os.path.join(base_dir, "model.tar.gz")
+
+    instance_count = ParameterInteger(name="InstanceCount", default_value=1)
+    instance_type = ParameterString(name="InstanceType", default_value="ml.m5.xlarge")
+
+    model = MXNetModel(
+        entry_point=entry_point,
+        source_dir=source_dir,
+        role=role,
+        model_data=mx_mnist_model_data,
+        framework_version="1.7.0",
+        py_version="py3",
+        sagemaker_session=pipeline_session,
+    )
+
+    step_args = model.register(
+        content_types=["*"],
+        response_types=["application/json"],
+        model_package_group_name=f"{pipeline_name}TestModelPackageGroup",
+        inference_instances=["ml.m5.xlarge"],
+        transform_instances=["ml.m5.xlarge"],
+        description="test-description",
+    )
+    step_model_regis = ModelStep(
+        name="mxnet-register-model",
+        step_args=step_args,
+    )
+
+    pipeline = Pipeline(
+        name=pipeline_name,
+        parameters=[instance_count, instance_type],
+        steps=[step_model_regis],
+        sagemaker_session=pipeline_session,
+    )
+
+    try:
+        pipeline.create(role)
+
+        for _ in retries(
+            max_retry_count=5,
+            exception_message_prefix="Waiting for a successful execution of pipeline",
+            seconds_to_sleep=10,
+        ):
+            execution = pipeline.start()
+            try:
+                execution.wait(delay=30, max_attempts=60)
+            except WaiterError:
+                pass
+            execution_steps = execution.list_steps()
+
+            assert len(execution_steps) == 1
+            failure_reason = execution_steps[0].get("FailureReason", "")
+            if failure_reason != "":
+                logging.error(
+                    f"Pipeline execution failed with error: {failure_reason}." " Retrying.."
+                )
+                continue
+            assert execution_steps[0]["StepStatus"] == "Succeeded"
+            if _REGISTER_MODEL_NAME_BASE in execution_steps[0]["StepName"]:
+                assert execution_steps[0]["Metadata"][_REGISTER_MODEL_TYPE]
+            break
+
+    finally:
+        try:
+            pipeline.delete()
+        except Exception as error:
+            logging.error(error)
+
+
+def test_model_registration_with_drift_check_baselines(
```

Comment:
test_model_registration_with_drift_check_baselines_and_model_metrics

Reply:
Updated in the new commit